Source/Wikitivity.Agent/WikitivityUploadsAgent.cs

```diff
@@ -51,155 +57,130 @@ public override void Execute()
-										var importSuccess = wAHelper.ImportDocument(this.Helper, workspaceID, fullyPopulatedArticleList, this);
+									if (importSuccess)
+									{
+										RaiseMessage("Import successful", 1);
+										// AT THIS POINT EVERYTHING IN THE BATCH SHOULD BE UPLOADED TO RELATIVITY
+										// 3. Delete the articles in the batch set from the RDO in the workspace
+										var deletionSuccess = wAHelper.DeleteUploadedRequestsFromRelativityOM(proxy, workspaceID, batchedRequestList);
 
-										if (importSuccess)
+										if (deletionSuccess)
 										{
-											RaiseMessage("Import successful", 1);
-											// AT THIS POINT EVERYTHING IN THE BATCH SHOULD BE UPLOADED TO RELATIVITY
-											// 3. Delete the articles in the batch set from the RDO in the workspace
-
-											var deletionSuccess =
-												wAHelper.DeleteUploadedRequestsFromRelativity(proxy, workspaceID,
-													batchedRequestList);
-
-											if (deletionSuccess)
+											// 4. Remove the entries put into the batch set from the master list & clear out the batch set
+											if (completeListByRequestID.Any())
 											{
-												// 4. Remove the entries put into the batch set from the master list & clear out the batch set
-												if (completeListByRequestID.Any())
-												{
-													completeListByRequestID.RemoveAll(item => batchedRequestList.Contains(item));
-													RaiseMessage("Removed batch set contents from master list", 1);
-												}
-												batchedRequestList.Clear();
+												completeListByRequestID.RemoveAll(item => batchedRequestList.Contains(item));
+												RaiseMessage("Removed batch set contents from master list", 1);
 											}
+											batchedRequestList.Clear();
 										}
+									}
 
-										// 6. Repopulate the batchset and repeat the process til the master list is empty. HANDLED BY THE WHILE LOOP
+									// 6. Repopulate the batchset and repeat the process til the master list is empty. HANDLED BY THE WHILE LOOP
 
-									}
-									catch (Exception ex)
-									{
-										RaiseError("Agent failed in execution: " + ex.ToString(), ex.ToString());
-									}
 								}
-							}
-							else
-							{
-								RaiseMessage("No jobs currently available", 1);
+								catch (Exception ex)
+								{
+									RaiseError("Agent failed in execution: " + ex.ToString(), ex.ToString());
+								}
 							}
 						}
-					}
-					else
-					{
-						RaiseMessage("No jobs currently available", 1);
+						else
+						{
+							RaiseMessage("No jobs currently available", 1);
+						}
 					}
 				}
+				else
+				{
+					RaiseMessage("No jobs currently available", 1);
+				}
+
 
 			}
 			catch (Exception ex)
 			{
-				RaiseError("Agent failed in execution: " + ex.ToString(), ex.ToString());
+				RaiseError("Agent failed in execution: " + ex.ToString(), ex.ToString() + "\r\n" + ex.InnerException + "\r\n" + ex.Message + "\r\n" + ex.Source);

```

Comment:
Typically you want to avoid this.  Raise a message that is readable  ex.Message.  Additionally, use logging to write the exception out